File: src/main/java/DataView/project/domain/Subject.java
package DataView.project.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Subject {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "timetable_id")
    private TimeTable timeTable;

    @ManyToOne
    @JoinColumn(name = "course_id")
    private Course course;

    @OneToOne(mappedBy = "subject", cascade = CascadeType.REMOVE)
    private Credit credits;

    @Column(name = "area")
    private String area; // 영역

    @Column(name = "sub_area")
    private String subArea; // 소영역

    @Column(name = "college")
    private String college; // 개설 단과 대학

    @Column(name = "department")
    private String department; // 개설 학과

    @Column(name = "subject_year")
    private String subjectYear; // 수강 학년

    @Column(name = "curriculum_type")
    private String curriculumType; // 교과 구분

    @Column(name = "course_code")
    private String courseCode; // 과목 코드

    @Column(name = "course_name")
    private String courseName; // 과목명

    @Column(name = "professor")
    private String professor; // 담당 교수

    @Column(name = "credit")
    private String credit; // 과목 학점

    @Column(name = "lecture_time")
    private String lectureTime; //강의 시간
}

File: FE/src/components/mypageMenu/timetable/Subject.js
import React from 'react'

const Subject = ({subject,onClickAddSubjectBtn}) => {
  return (
    <>
      <div className='subject'>
        <div className='subjectInfo'>
          <h4>{subject.courseName}</h4>
          <h5>{subject.professor}</h5>
          <h6>{subject.lectureTime}</h6>
          <div>
            <h6>{subject.subjectYear}학년</h6>
            <h6>{subject.area}</h6>
            <h6>{subject.credit}학점</h6>
            <h6>{subject.courseCode}</h6>
          </div>
        </div>
        <div className='addSubjectBtn_area'>
          <button
            onClick={()=>{onClickAddSubjectBtn(subject.id)}} 
            className='addSubjectBtn'>추가하기</button>
        </div>
      </div>
    </>
  )
}

export default Subject